src/Symfony/Component/AstGenerator/AstGeneratorChain.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * Generator delegating the generation to a chain of generators.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class AstGeneratorChain implements AstGeneratorInterface
```

Comment:
> But for the routing generators i would like to allow the user to easily hook into them and it could pass by inheritance.

How? Those classes have only `private` members and 2 public methods. And those 2 public methods could be easily changed with composition if developer really needs it.
This is weird Symfony internal rule: to keep members `private` by default, but in the same time do not require `final`: there is no point to open them for inheritance then.